common/src/main/java/org/apache/sedona/common/raster/MapAlgebra.java

```diff
@@ -437,6 +437,57 @@ public static double[] normalize(double[] band) {
         return result;
     }
 
+    /**
+     *
+     * @param rasterGeom Raster to be normalized
+     * @return a raster with all values in all bands normalized between [0 - 255]
+     */
+    public static GridCoverage2D normalizeAll(GridCoverage2D rasterGeom) {
+        return normalizeAll(rasterGeom, 0d, 255d);
+    }
+
+    /**
+     *
+     * @param rasterGeom Raster to be normalized
+     * @param minLim Lower limit of normalization range
+     * @param maxLim Upper limit of normalization range
+     * @return a raster with all values in all bands normalized between minLim and maxLim
+     */
+    public static GridCoverage2D normalizeAll(GridCoverage2D rasterGeom, double minLim, double maxLim) {
+        if (minLim > maxLim) {
+            throw new IllegalArgumentException("minLim cannot be greater than maxLim");
+        }
+
+        int numBands = rasterGeom.getNumSampleDimensions();
+
+        for (int bandIndex = 1; bandIndex <= numBands; bandIndex++) {
+            // Get the band values as an array
+            double[] bandValues = bandAsArray(rasterGeom, bandIndex);
+
+            // Find min and max values in the band
+            double minValue = Arrays.stream(bandValues).min().orElse(Double.NaN);
+            double maxValue = Arrays.stream(bandValues).max().orElse(Double.NaN);
+            System.out.println("minValue: "+minValue);
+            System.out.println("maxValue: "+maxValue);
+
+            if (minValue == maxValue) {
```

Comment:
Please add 2 more optional parameters to this function which takes the minValue and maxValue from users. If the 2 values are not given, we compute the min and max ourselves.

Reply:
@jiayuasu Just curious, why is taking minValue and MaxValue necessary? The current implementation i'm working on is able to compute minValue and maxValue excluding the NoDataValue.

Reply:
There will be cases that users know the expected value range of all band data in advance. In this case, finding the min max values ourselves does not make sense.

Reply:
To handle all pixel data types, would having a pixelType argument like in RS_MapAlgebra help?

Reply:
No need. Because this NormalizeAll function is not supposed to change the original pixel data type.